### ISSUE TITLE
Updating NVDA leaves empty folders behind

### ISSUE THREAD

**Issue by SeanFarrow** [nvaccess/nvda#9363]
Updating NVDA from 2018.4 to 2018.4.1 leaves an empty directory for 2018.4 in both the lib and lib64 directories.

I'd be happy to look at fixing this but am unsure whether the problem should be fixed in the installer or the update checking/installation mechanism.

**Comment by Brian1Gaff** [comment]
Hi this happens as attempts are made to remove them but they do not always 
work. In the early days this threw an error and stopped nvda  from 
restarting. Now if you look at the log it often says the items failed to be 
removed would be attempted next time. Whether this is successful, I'd say 
yes, but then often you get the same effect with the previous folders . 
Renaming and removing can be a little sluggish in Windows if you are doing a 
lot of it in the process.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by SeanFarrow** [comment]
Hi,

Perhaps if this fails during the update NVDA should remove the folder on next start?

**Comment by Brian1Gaff** [comment]
I'm not sure if this is done or not. I do know it was talked about at the 
time this issue came up but your best bet is to set the log to debug and see 
what actually happens on an update and the next start, cannot say I've seen 
any activity till the next update. Do empty folders really matter at all? 
Lots of software uninstallers leave their folders behind, not always empty. 
Did somebody mention google? :-)
Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by SeanFarrow** [comment]
I can confirm that it doesn’t remove the folders on next.

Yes, empty folders do matter as it shows a problem ith the installer itself. The installer installs the folder, so in my view should remove it.

**Comment by Adriani90** [comment]
cc: @lukaszgo1 I recall there was a PR related to this in the past but I am not sure if it was from you. Maybe you find some time to look into this.
